test/controllers/NumberOfStatLeaveDaysControllerSpec.scala

```diff
@@ -0,0 +1,782 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import java.time.LocalDate
+
+import base.SpecBaseControllerSpecs
+import forms.NumberOfStatLeaveDaysFormProvider
+import messages.NumberOfStatLeaveDaysMessages
+import models.EmployeeStarted
+import models.requests.DataRequest
+import org.mockito.Matchers.any
+import org.mockito.Mockito.when
+import org.scalatestplus.mockito.MockitoSugar
+import pages._
+import play.api.data.Form
+import play.api.mvc.AnyContentAsEmpty
+import play.api.test.CSRFTokenHelper._
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import utils.LocalDateHelpers.{apr5th2020, apr6th2019, apr6th2020, feb1st2020}
+import viewmodels.{BeenOnStatutoryLeaveHelper, NumberOfStatLeaveDaysHelper}
+import views.ViewUtils.dateToString
+import views.html.NumberOfStatLeaveDaysView
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class NumberOfStatLeaveDaysControllerSpec extends SpecBaseControllerSpecs with MockitoSugar {
+
+  val formHelper    = app.injector.instanceOf[NumberOfStatLeaveDaysHelper]
+  val contentHelper = app.injector.instanceOf[BeenOnStatutoryLeaveHelper]
+  val view          = app.injector.instanceOf[NumberOfStatLeaveDaysView]
+  val postAction    = controllers.routes.NumberOfStatLeaveDaysController.onSubmit()
+  val formProvider  = new NumberOfStatLeaveDaysFormProvider()
+
+  def form(boundaryStart: LocalDate, boundaryEnd: LocalDate): Form[Int] = formProvider(boundaryStart, boundaryEnd)
+
+  lazy val numberOfDaysOnStatutoryLeaveRoute = routes.NumberOfStatLeaveDaysController.onPageLoad().url
+
+  val getRequest: FakeRequest[AnyContentAsEmpty.type] =
+    FakeRequest(GET, numberOfDaysOnStatutoryLeaveRoute).withCSRFToken
+      .asInstanceOf[FakeRequest[AnyContentAsEmpty.type]]
+
+  val controller = new NumberOfStatLeaveDaysController(
+    messagesApi = messagesApi,
+    sessionRepository = mockSessionRepository,
+    navigator = navigator,
+    identify = identifier,
+    getData = dataRetrieval,
+    requireData = dataRequired,
+    formProvider = formProvider,
+    formHelper = formHelper,
+    contentHelper = contentHelper,
+    controllerComponents = component,
+    view = view
+  )
+
+  "NumberOfDaysOnStatLeave Controller" when {
+
+    "onPageLoad" when {
+
+      "employee is type 3" when {
+
+        "the day before employee is first furloughed is before the 5th April 2020" must {
+
+          "return OK and the correct value for a GET" in {
+
+            val firstFurloughDate = LocalDate.parse("2020-04-05")
+            val furloughStartDate = LocalDate.parse("2021-03-01")
+            val boundaryStartDate = apr6th2019
+            val boundaryEndDate   = firstFurloughDate.minusDays(1)
+            val boundaryStart     = dateToString(apr6th2019)
```

Comment:
This is picky but could this `val boundaryStart` be `val boundaryStart = dateToString(boundaryStartDate)`
and similar with boundaryEnd.

Reply:
no that's fair makes it more readable I'll make the change and go through the others